pkg/runtime/reconciler.go

```diff
@@ -271,10 +272,17 @@ func (r *resourceReconciler) Sync(
 		return err
 	}
 	for _, condition := range latest.Conditions() {
-		if condition.Type == ackv1alpha1.ConditionTypeResourceSynced &&
-			condition.Status != corev1.ConditionTrue {
-			return requeue.NeededAfter(
-				ackerr.TemporaryOutOfSync, requeue.DefaultRequeueAfterDuration)
+		if condition.Type == ackv1alpha1.ConditionTypeResourceSynced {
+			if condition.Status == corev1.ConditionTrue {
+				duration, isRequeueOnSuccess := r.rmf.GetRequeueOnSuccessSeconds()
+				if isRequeueOnSuccess {
+					return requeue.NeededAfter(
+						ackerr.RequeueOnSuccess, (time.Duration(duration) * time.Second))
+				}
```

Comment:
If you take my suggestions above and below, this should be a little easier:

```go
if condition.Status == corev1.ConditionTrue {
    if duration := r.rmf.RequeueOnSuccessSeconds(); duration > 0 {
        return requeue.NeededAfter(nil, time.Duration(duration)*time.Second)
    }
}
```

Reply:
yep, cleaner code :)